### ISSUE TITLE
Composite PK support?

### ISSUE THREAD

**Issue by pdhimate** [jsakamoto/EntityFrameworkCore.IndexAttribute#10]
Can you add support for Composite PK?
E.g.

[PrimaryKey("PK_AtoB", 0, IsClustered = true)]
public long Id1 { get; set; }
[PrimaryKey("PK_AtoB", 1, IsClustered = false)]
public long Id2 { get; set; }

Creates a table with clustered Id1 and non clustered Id2, with Pk and { Id1, Id2 }  , both.

Currently, as a work around, I manually update the migration to:
  table.PrimaryKey("PK_AtoB", x => new { x.Id1, x.Id2})
                        .Annotation("SqlServer:Clustered", true);
                  

In EF 6.x this was acheivable by
[Key, Column(Order = 0)]
public long Id1 { get; set; }
[Key, Column(Order = 1)]
public long Id2 { get; set; }

**Comment by jsakamoto** [comment]
Of course, `Toolbelt.EntityFrameworkCore.IndexAttribute.SqlServer` already supports the composite primary key.

If you annotate the model like this,

```csharp
[PrimaryKey("PK_AtoB", 0, IsClustered = true)]
public long Id1 { get; set; }

[PrimaryKey("PK_AtoB", 1, IsClustered = true)]
public long Id2 { get; set; }
```

then the migration code will be as below.

```csharp
table.PrimaryKey("PK_AtoB", x => new { x.Id1, x.Id2})
    .Annotation("SqlServer:Clustered", true);
```

However, your question,

[PrimaryKey("PK_AtoB", 0, IsClustered = **true**)]
public long Id1 { get; set; }
[PrimaryKey("PK_AtoB", 1, IsClustered = **false**)]
public long Id2 { get; set; }

is no make sense.

Because the "Clustered" or "NonClustered" is an attribute that belongs to Index itself, not each column.

At this time, no error occurs even if you specify conflicted annotations like above, but I started to plan to make it causes a runtime error.

**Comment by pdhimate** [comment]
Got it. What I meant was having a composite non-clustered PK and a clustered index on Id1, but I think your syntax here will work for my case. Thanks!

**Comment by jsakamoto** [comment]
> What I meant was having a composite non-clustered PK and a clustered index on Id1,

Yes, if you annotate like this, 

```csharp
[Index("Index_Id1", IsClustered = true)]
[PrimaryKey("PK_AtoB", 0, IsClustered = false)]
public long Id1 { get; set; }

[PrimaryKey("PK_AtoB", 1, IsClustered = false)]
public long Id2 { get; set; }
```

then you will get this migration code.

```csharp
migrationBuilder.CreateTable(
    ...
    constraints: table =>
    {
        table.PrimaryKey("PK_AtoB", x => new { x.Id1, x.Id2 })
            .Annotation("SqlServer:Clustered", false);
    });

migrationBuilder.CreateIndex(
    name: "Index_Id1",
    table: "...",
    column: "Id1")
    .Annotation("SqlServer:Clustered", true);
```
